### ISSUE TITLE
Avoiding duplication

### ISSUE THREAD

**Issue by philippjfr** [gngdb/holo-nets#5]
This looks great and is a good start for interfacing with HoloViews. Note however that HoloViews already has a class devoted to running a simulation/model, collecting data and then save it to a pickle. We're currently still working on the functionality to load that data back in but in general the system would be something like this.
- Setup: You set up a so called Collector by supplying it with a `run` function, a number of measurements and some deferred analyses. For a Topographica model this looks something like [this](http://ioam.github.io/topographica/Tutorials/GCAL_Collector.html). The Collector will then call the run function with the desired number of epochs that are to be run, then execute all the "measurements",  which can be processed further with some analyses, next run the next set of epochs and repeat until it's done. The Collector can be run either directly in the notebook or you can write a trivial subclass or wrapper class to run it in batch mode and save the collected data out (presumably as pickles) - here's the [Topographica BatchCollector](https://github.com/ioam/topographica/blob/fe0c5d66e8e66cd8376566c602a65163d481f0a0/topo/misc/lancext.py) as an example.
- Running: You can either call this Collector directly or use [Lancet](ioam.github.io/lancet) to parameterize your model so you can dispatch multiple instances to be run either in sequence or in parallel. Lancet will then take care of running all the jobs, logging lots of metadata (parameters, version control, library versions) and placing the resulting files in their own timestamped folders.
- Loading: This part is not completely worked out yet, but I'll be working on it over the next few days. The basic idea is that Lancet provides an easy way to load the metadata and filepaths of your runs. Once you have them, they can be loaded into a HoloViews object. Say you've run networks with 4x4, 8x8 and 12x12 kernels and recorded data every 10 epochs. You'll then be able to select the epochs and kernel size you want to load data for, and collate all the selected data. Once everything is loaded you can then compare how your measurements, e.g. your weights, evolve over the epochs or how they compare across kernel sizes. This is a slight simplification as you'll also be able to select which measurements you want to load, e.g. load just the weights or just the error curves.

Let me know if any of this doesn't make sense to you, or if you think any of this workflow is unsuitable for neural nets.

**Comment by gngdb** [comment]
Sounds like some of this code will just be replaced by Holoviews itself
then; I'll have to take a closer look at Collectors. Some of it is code to
run the experiment. Guess it's just important to keep a clear distinction
between code interfacing Holoviews and code needed to run the experiment.
I'll still bring this to version 0.1, but after that will change direction
of this project to link better with Holoviews/Lancet.
